pkg/test/tls.go

```diff
@@ -0,0 +1,209 @@
+tINuOy990fr1BkhzSoFD/xVhQMqoDN00+B8mIWJ9pstknDcAM+6Z+FfUuwzLcxx5
+LogpOIGhYt+03ywXDxfuvXfRFXFGCwifcLx8XyiOchBckxLtfXioC4Q61gWE63VN
+J7qyPDrdUZ/CCY5Vz+qRqsF2rlwOWPluXrWzaKzlvmCm5iQ8J58c6dHhyY8Yob8K
+xHo9F56bAgMBAAECggEARgN6Hg+3FwRio4SoPnWoCErghM8yOC1MRs5013C82HAm
+m5Q6l5+YQbTiJ3f4kFmNz2gYhucYZUOS2kUPVQ2kWbfhFEO4aHt/n0+s1wUKH5yG
+PDP000VX8fVDdGtzUYJy4VZvmMhQ/M6s/wQr+eALeHALFmztAgXiGIemJPBZeu+c
+rbAFowzdYsHpTSrh2nm0HYSoyZr87wQoQjLo0c6FqLLxYcnCvMemcg8HK4iw6I0a
+vbbG61Pdg4CNy6ZgSS96+WwiixrcWoQ7CN339giRmFT2e6vulbreQtD4QzlmqtWW
+su+Du3kc/9XqtJwOpyOkHyXrALMnpPC+c0WnVp76xQKBgQDa/kd79kOWzQXTKmqn
+heuaKq/UTMYlGhc13KhEpXeX/r0skWMKjuvxgVacNxv4F4d2axCLmADh1UotZsDz
+9Hlvq5aPn8f/VYv0CCwOrigSyv25AEAZCnk84igbiSq1TfXppJzyqYoLue2HrW4k
+CHL9KviBwauePNRDv+KNZJGALQKBgQDBVLHMDF6xjsy27T4toqLz1thdiBjd/GR+
+SE6+j18z/CKJVCWlRBykmJ3vxW9qAptdb86ZxmfIFnkWJTt/yEthvyfqHe6ksXED
+SElOKuReUYAPJazaGS5F9TywGnsVxhwn9BtqPMy/b7WRQJowcG9zSlntjT+cn7sg
+GfSB6ASO5wKBgDaiRXc5ovcWQyPBa0ZL9NFLYP5YAP70mWHIoPovRbzXwp5BzzGt
+IlPn7pGedg3Y4OS8JS6OR3oP2ielgPHbxggECNXgCOc8kmPZPhSTgk/d8Jqc42Db
+6g80ZMkp2UvOHVGizb0Eavot8oJs1BONQBLFC6ZjiMs7ZcFZN84KjvopAoGAYPAz
+ummVbZh5o1tv6vf6lyNqF/Pu7Bfq17sv6LMA/JL3Sj6sJaLybcGsp5Yq2E/4UTCH
+umlWjmheTLFclST8T0XHIMfjaici0I+FWjF9kqFxAadVdYJcxm1CAdc1UmSkp4/p
+0yorS+4ab3uiFJm7+GYWk1tYwxMAhAcfp6eL6Y8CgYAMNptHbD/fL0wrXDBVWRYP
+LL55dHKf6Op+c2N9/82WsaMpHQ/A8MiIn2iwQReEIjl0zk+OfvTgQEz1FEw+zZk/
+kFWRLF8Hel5mv0JKz8ExZ4LHTTE8OpxvEbvP3j6A/MngpsuN73LA4i/GqrktGtiy
+UYF0gdXCSiPfvXct5lwU7g==
+-----END PRIVATE KEY-----`
+
+// Server cert generated with:
+// openssl x509 -req -days 3650 -sha256 -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt -extfile <(echo subjectAltName = IP:127.0.0.1)
+const serverCert string = `-----BEGIN CERTIFICATE-----
+MIIDXzCCAkegAwIBAgIUZGbRxIm/HK0OiVdTzIFWmnPqttUwDQYJKoZIhvcNAQEL
+BQAwDDEKMAgGA1UEAwwBKjAeFw0yMzA4MDIwODU0NTlaFw0zMzA3MzAwODU0NTla
+MHIxCzAJBgNVBAYTAkdCMQ8wDQYDVQQIDAZMb25kb24xDzANBgNVBAcMBkxvbmRv
+bjEdMBsGA1UECgwUbGlidmF1bHQgY29uc3VsdGFudHMxFjAUBgNVBAsMDUlUIERl
+cGFydG1lbnQxCjAIBgNVBAMMASowggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEK
+AoIBAQClYidsRfhbhwxxGUZ7X7/CwltfZCof/FmrYqzsWzimr8tUWPy/AHJHo+A0
+8ixuZMBuCn2Uh22gFaEZKn3Uk8ZmGXb9IhYZxTJrYx9+ZUvBaN5g8072bbF2TCCD
+YhX1w9o488mSVe3vP/I/tINuOy990fr1BkhzSoFD/xVhQMqoDN00+B8mIWJ9pstk
+nDcAM+6Z+FfUuwzLcxx5LogpOIGhYt+03ywXDxfuvXfRFXFGCwifcLx8XyiOchBc
+kxLtfXioC4Q61gWE63VNJ7qyPDrdUZ/CCY5Vz+qRqsF2rlwOWPluXrWzaKzlvmCm
+5iQ8J58c6dHhyY8Yob8KxHo9F56bAgMBAAGjUzBRMA8GA1UdEQQIMAaHBH8AAAEw
+HQYDVR0OBBYEFOdbqIlktmQGpRr9ydKkACwx/OnhMB8GA1UdIwQYMBaAFDfhKgNf
+8sL6LaHntwho4aYZlOAHMA0GCSqGSIb3DQEBCwUAA4IBAQAMKfFetofRb9dFInU5
+KpF3+IVwrR53UbUbNF0mnQ7aNRE7YfLPRTOV2Dp5zeOlUiO6FhK1AkCcs1RILzUM
+bUwolEbgQRmMV8NPyY+0vkBQDJQYfw3bHm2NCWRKd2A0KI9rX1VpWvY3Z300zmLM
+TPgRGwN4oZbQLpbI6iZ+MuaBw9c3xOuVKGI0OQybl7MM49Uk/QAf+Ltb+VD/b+NR
+QtOnsqqqb3s8LlqTbYn1zM9FSX2YNRljDkElTVfzhlD2qpMvy8Ep8qrAlFcI8yZ8
+HKUIvMe6pjPWHHGVkKBldRqQIOH5WoUSKjrC8koV+Kqj6PMXKquyZdvdC3bhgj4l
+Pnib
+-----END CERTIFICATE-----`
```

Comment:
Should we also try with an invalid cert ?

Reply:
done, I've added a test where the client certificate is signed by an authority unknown from the server